src/probnum/diffeq/odefiltsmooth/ivpfiltsmooth.py

```diff
@@ -166,175 +162,269 @@ def init_implementation(f, y0, t0, prior, initrv, df=None):
+        )
+        full_meas_cov = full_meas_cov_cholesky @ full_meas_cov_cholesky.T
+        meas_rv = Normal(
+            mean=meas_rv_error_free.mean,
+            cov=full_meas_cov,
+            cov_cholesky=full_meas_cov_cholesky,
         )
 
-        # 4. Update
-        zero_data = np.zeros(meas_rv.mean.shape)
-        filt_rv, _ = self.gfilt.measurement_model.backward_realization(
-            zero_data, pred_rv, rv_forwarded=meas_rv, gain=info["gain"]
+        # Estimate local diffusion_model and error
+        diffusion_error_free = self.diffusion_model.calibrate_locally(
+            meas_rv_error_free
+        )
+        diffusion_full = self.diffusion_model.calibrate_locally(meas_rv)
+        diffusion_for_calibration = self.diffusion_model.update_current_information(
+            diffusion_full, diffusion_error_free, t_new
         )
+        local_errors = np.sqrt(diffusion_for_calibration) * np.sqrt(
+            np.diag(meas_rv_error_free.cov)
+        )
+
+        # Either re-predict and re-measure with improved calibration or let the predicted RV catch up.
+        if self._repeat_after_calibration:
+            pred_rv, _ = self.dynamics_model.forward_rv(
+                rv=current_rv, t=t, dt=dt, _diffusion=diffusion_for_calibration
+            )
+            meas_rv, info = self.measurement_model.forward_rv(
+                rv=pred_rv, t=t_new, compute_gain=True
+            )
+            gain = info["gain"]
+        else:
+            # Combine error-free and noisy-component predictions into a full prediction.
+            # (The measurement has been updated already.)
+            full_pred_cov_cholesky = utils.linalg.cholesky_update(
+                pred_rv_error_free.cov_cholesky, pred_sqrtm
+            )
+            full_pred_cov = full_pred_cov_cholesky @ full_pred_cov_cholesky.T
+            pred_rv = Normal(
+                mean=pred_rv_error_free.mean,
+                cov=full_pred_cov,
+                cov_cholesky=full_pred_cov_cholesky,
+            )
 
-        # 5. Error estimate
-        local_errors = self._estimate_local_error(
-            pred_rv,
-            t_new,
-            self.sigma_squared_mle * proc_noise_cov,
-            np.sqrt(self.sigma_squared_mle) * proc_noise_cov_cholesky,
+            # Gain needs manual catching up, too.
+            crosscov = full_pred_cov @ H.T
+            gain = scipy.linalg.cho_solve((meas_rv.cov_cholesky, True), crosscov.T).T
+
+        # 4. Update
+        zero_data = np.zeros(meas_rv.mean.shape)
+        filt_rv, _ = self.measurement_model.backward_realization(
+            zero_data, pred_rv, rv_forwarded=meas_rv, gain=gain
         )
-        err = np.linalg.norm(local_errors)
 
-        return filt_rv, err
+        proj = self.dynamics_model.proj2coord(coord=self._reference_coordinates)
+        reference_state = np.maximum(proj @ filt_rv.mean, proj @ current_rv.mean)
+        return filt_rv, local_errors, reference_state
```

Comment:
Do we have an alternative to "state"? In my mind, "state" describes the d*(q+1)-dimensional object of the full random process, not a projection. But if we don't have a nice alternative,  "state" is fine :)